src/libraries/System.ComponentModel.Composition/src/System/ComponentModel/Composition/ReflectionModel/LazyMemberInfo.cs

```diff
@@ -97,14 +97,13 @@ public override int GetHashCode()
             {
                 return MemberType.GetHashCode() ^ _accessorsCreator.GetHashCode();
             }
-            else
+
+            if (_accessors != null && _accessors[0] is MemberInfo accessor)
             {
-                if (_accessors == null || _accessors[0] == null)
-                {
-                    throw new Exception(SR.Diagnostic_InternalExceptionMessage);
-                }
-                return MemberType.GetHashCode() ^ _accessors[0]!.GetHashCode(); // TODO-NULLABLE: Indexer nullability tracked (https://github.com/dotnet/roslyn/issues/34644)
+                return MemberType.GetHashCode() ^ accessor.GetHashCode();
```

Comment:
nit: should this use hash code combiner? cc: @GrabYourPitchforks

Reply:
`HashCodeCombiner` is good for things that don't have a great distribution already. One example is a record type where the user is responsible for providing most of the fields, which is likely to see many of the values clustered around the same set of hash codes. Since this hash code calculation routine is using reflection and other runtime objects, there's not as much concern about clustering, since these objects already have good distribution.